src/inference/inference.ts

```diff
@@ -3,8 +3,18 @@ import {
   EmbedRequestInputsInner,
   EmbedRequestParameters,
   InferenceApi,
+  Rerank200Response as RerankResponse,
 } from '../pinecone-generated-ts-fetch/inference';
 import { EmbeddingsList } from '../models';
+import { PineconeArgumentError } from '../errors';
+import { prerelease } from '../utils/prerelease';
+
+export interface RerankOptions {
```

Comment:
We should make sure that this type and the generated types that are used for inference are exported from the top level of the package so people have access to them: https://github.com/pinecone-io/pinecone-ts-client/blob/73d98b8aaffd93dafadab8902f6728b153ae25a9/src/index.ts#L45-L63

I'm guessing there might be a bunch of embedding / rerank stuff that we're using in these functions and maybe also need to be exposed.

Reply:
So, for inputs, the users shouldn't need the exported types or interfaces, since we wrap those in more ux-friendly stuff. But I've now added the rerank response types to this file, good call. (The response type `EmbeddingsList` for /embed is wrapped, and we export that.

Reply:
Yup, thanks!